sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java

```diff
@@ -504,6 +505,14 @@ public UnresolvedExpression visitScoreRelevanceFunction(ScoreRelevanceFunctionCo
 
   private Function buildFunction(String functionName,
                                  List<FunctionArgContext> arg) {
+    if (functionName.equals("nested")
```

Comment:
We should be able to add the `NestedAllFields` expression without altering this function and using standard visitor pattern.